src/Instrument/Transformer/FilterInjectorTransformer.php

```diff
@@ -39,30 +40,36 @@ class FilterInjectorTransformer implements SourceTransformer
     protected static $options = array();
 
     /**
+     * @var AspectKernel|null
+     */
+    protected static $kernel;
+
+    /**
      * Class constructor
      *
-     * @param array $options Configuration options from kernel
+     * @param AspectKernel $kernel Kernel to take configuration from
      * @param string $filterName Name of the filter to inject
      */
-    public function __construct(array $options, $filterName)
+    public function __construct(AspectKernel $kernel, $filterName)
     {
-        self::configure($options, $filterName);
+        self::configure($kernel, $filterName);
     }
 
     /**
      * Static configurator for filter
      *
-     * @param array $options Kernel options
+     * @param AspectKernel $kernel Kernel to use for configuration
      * @param string $filterName Name of the filter to inject
      *
      * @throws \RuntimeException if filter was configured early
      */
-    protected static function configure(array $options, $filterName)
+    protected static function configure(AspectKernel $kernel, $filterName)
     {
-        if (self::$options) {
+        if (!is_null(self::$kernel)) {
```

Comment:
Do not use `is_null()` check, PHP is dynamic language, so `if (!self::$kernel)` will be better for me.